modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ScriptProcessor.java

```diff
@@ -120,16 +117,17 @@ public ScriptProcessor create(
                 Arrays.asList("id", "source", "inline", "lang", "params", "options").forEach(config::remove);
 
                 // verify script is able to be compiled before successfully creating processor.
-                IngestScript ingestScript = null;
+                IngestScript.Factory ingestScriptFactory = null;
                 try {
-                    final IngestScript.Factory factory = scriptService.compile(script, IngestScript.CONTEXT);
-                    if (ScriptType.INLINE.equals(script.getType())) {
-                        ingestScript = factory.newInstance(script.getParams());
+                    ingestScriptFactory = scriptService.compile(script, IngestScript.CONTEXT);
+                    if (ScriptType.STORED.equals(script.getType())) {
+                        // do not cache stored scripts lest they change and invalidate the cached value
```

Comment:
Is this a change in current behavior? It might be worth pulling this out into a separate PR, only so that it can be noted (perhaps as a bugfix) in the release notes, whereas if it goes in within this enhancement/addition to the fields API it would be hidden.

Reply:
This is the same as current behavior, it failed a test which is how I found it.

Reply:
This is the current behavior, it's just the instance used to be cached.  This change is due to a failed unit test when I switched to the compiled script rather than the instance.